### PR TITLE
Merge back Release 1.2.1

### PR DESCRIPTION
**Pull request by rolandlo**
I'm not sure if I did it right with the merge, since I had trouble with the release-helper script (for the merge back step) and did it manually. The total diff should be mostly fine. The changes to the glade file must be fixed though. 
Is the commit tree alright?

### CODE REVIEW

File: src/core/gui/inputdevices/PenInputHandler.cpp [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #include "PenInputHandler.h"
 
 #include <algorithm>  // for max, min
+#include <cassert>    // for assert
```
Comment (bhennion) [inline src/core/gui/inputdevices/PenInputHandler.cpp:8]:
```suggestion
#include "util/Assert.h"  // for xoj_assert
```
(+ associated clang-formatting)

File: src/core/gui/inputdevices/PenInputHandler.cpp [hunk 2]
```diff
@@ -171,46 +179,57 @@ auto PenInputHandler::actionStart(InputEvent const& event) -> bool {
...
+    }
 
-    if (filteredPressure != Point::NO_PRESSURE) {
-        filteredPressure *= settings->getPressureMultiplier();
-        filteredPressure = std::max(settings->getMinimumPressure(), filteredPressure);
+    double filteredPressure;
+    if (pressureMode == PressureMode::INFERRED_PRESSURE) {
+        filteredPressure = inferPressureValue(pos, page);
+    } else {
+        assert(pressureMode == PressureMode::DEVICE_PRESSURE);
```
Comment (bhennion) [inline src/core/gui/inputdevices/PenInputHandler.cpp:217]:
```suggestion
        xoj_assert(pressureMode == PressureMode::DEVICE_PRESSURE);
```

File: src/core/gui/inputdevices/PenInputHandler.cpp [hunk 3]
```diff
@@ -171,46 +179,57 @@ auto PenInputHandler::actionStart(InputEvent const& event) -> bool {
...
+            filteredPressure = lastPressure;
+        } else {
+            filteredPressure = pos.pressure;
+            lastPressure = pos.pressure;  // Record in case the pressure value is omitted in the next event
+        }
     }
 
-    return filteredPressure;
+    Settings* settings = this->inputContext->getSettings();
+    assert(settings->getMinimumPressure() >= 0.01);
```
Comment (bhennion) [inline src/core/gui/inputdevices/PenInputHandler.cpp:231]:
```suggestion
    xoj_assert(settings->getMinimumPressure() >= 0.01);
```

**Review by bhennion — CHANGES_REQUESTED** [review]
The assert()'s need to become xoj_assert()'s. I did not check the .glade file yet. Do you want me to have a look?

**Comment by rolandlo** [comment]
> The assert()'s need to become xoj_assert()'s. 

Right!

> I did not check the .glade file yet. Do you want me to have a look?

Yes, please! Something with the `GtkAlignment`s seems wrong.

**Comment by bhennion** [comment]
> Yes, please! Something with the `GtkAlignment`s seems wrong.

It was a mess indeed. I corrected it and forced pushed. (I did not take care of the assert() though)

**Comment by rolandlo** [comment]
Should commit [1ac6814](https://github.com/xournalpp/xournalpp/pull/5090/commits/1ac68146637660708837661f9c357f56f9777d55) be really part of the merge? In #4455 we didn't have the version bump commit included.

File: ui/settings.glade [hunk 4]
```diff
@@ -5469,6 +5470,77 @@ Set this to 0% to always re-render on zoom.</property>
                             <property name="orientation">vertical</property>
                             <child>
                               <object class="GtkFrame" id="sid169">
+                                                                  <property name="visible">True</property>
+                                <property name="can-focus">False</property>
```
Comment (bhennion) [inline ui/settings.glade:5474]: I f'ed up the indentation here:
```suggestion
                                <property name="visible">True</property>
                                <property name="can-focus">False</property>
```

**Comment by bhennion** [comment]
> Should commit [1ac6814](https://github.com/xournalpp/xournalpp/pull/5090/commits/1ac68146637660708837661f9c357f56f9777d55) be really part of the merge? In #4455 we didn't have the version bump commit included.

I don't know. It seems odd to me to take a commit out. @xournalpp/core what's the procedure here?

**Comment by Febbe** [comment]
Master requires a different version bump, therefore, it would be fine to leave that commit out. But we can also override it in the merge commit. Probably this is cleaner for the history. We will get merge conflicts any way.

**Comment by rolandlo** [comment]
In https://github.com/xournalpp/xournalpp/pull/3599#issuecomment-975509184 when merging back release 1.1.1 @LittleHuba requested to drop the version bump commit which would correspond to https://github.com/xournalpp/xournalpp/pull/5090/commits/eb070c46acb9badbd22635976aeffe97c1defbbf here. @LittleHuba can you explain, which version bump commits should be included and which shouldn't and for what reason?
Also how would I drop a commit in this merge (preferably without solving all merge conflicts again)? Using `git rebase -i --rebase-merges` ?

**Comment by rolandlo** [comment]
I have dropped the version bump commit to `1.2.2~dev` as outlined above and have fixed the indentation in the `.glade` file. From my side I would suggest to merge this from the CLI (with `ff-only`).

**Review by bhennion — APPROVED** [review]

**Comment by LittleHuba** [comment]
> In https://github.com/xournalpp/xournalpp/pull/3599#issuecomment-975509184 when merging back release 1.1.1 @LittleHuba requested to drop the version bump commit which would correspond to https://github.com/xournalpp/xournalpp/pull/5090/commits/eb070c46acb9badbd22635976aeffe97c1defbbf here. @LittleHuba can you explain, which version bump commits should be included and which shouldn't and for what reason?
> ~Also how would I drop a commit in this merge (preferably without solving all merge conflicts again)? Using `git rebase -i --rebase-merges` ?~
> 
> For the record: I can drop the last version bump commit using 
> ```term
> git rebase -i -r -X ours HEAD~<number of commits>
> git diff -R origin/<branchname> | git apply
> ``` 
> followed by amending the changes.
> 

Hi
Sorry that was way too long ago for me to remember in detail. I think the request came originally from a user. You could check out the last version bump PR for comments.

**Pull request merged**